src/main/scala/io/citrine/lolo/bags/Bagger.scala

```diff
@@ -114,65 +109,35 @@ case class Bagger(
-    for the training points. If a point has 0 uncertainty, the ratio is 1 if error is also 0, otherwise infinity */
-    val ratio = if (uncertaintyCalibration && isRegression && useJackknife) {
-      Async.canStop()
-      oobErrors.map {
-        case (_, 0.0, 0.0) => 1.0
-        case (_, _, 0.0) => Double.PositiveInfinity
-        case (_, error, uncertainty) => Math.abs(error / uncertainty)
-      }.sorted.drop((oobErrors.size * 0.68).toInt).head
-    } else {
-      1.0
-    }
-    assert(!ratio.isNaN && !ratio.isInfinity, s"Uncertainty calibration ratio is not real: $ratio")
-
     /* Wrap the models in a BaggedModel object */
-    if (biasLearner.isEmpty || oobErrors.isEmpty) {
+    val helper = BaggerHelper(models, trainingData, Nib, useJackknife, uncertaintyCalibration)
+    val biasModel = if (biasLearner.isDefined && helper.oobErrors.nonEmpty && helper.isRegression) {
       Async.canStop()
-      if (isRegression) {
-        new BaggedTrainingResult(models.asInstanceOf[ParSeq[Model[PredictionResult[Double]]]], averageImportance, Nib, trainingData, useJackknife, None, ratio, disableBootstrap)
-      } else {
-        new BaggedTrainingResult(models, averageImportance, Nib, trainingData, useJackknife, None, ratio, disableBootstrap)
-      }
+      Some(biasLearner.get.train(helper.biasTraining).getModel().asInstanceOf[Model[PredictionResult[Double]]])
+    } else None
+
+    if (helper.isRegression) {
+      new BaggedTrainingResult[Double](
+        models = models.asInstanceOf[ParSeq[Model[PredictionResult[Double]]]],
+        featureImportance = averageImportance,
+        Nib = Nib,
+        trainingData = trainingData,
+        useJackknife = useJackknife,
+        biasModel = biasModel,
+        rescale = helper.ratio,
+        disableBootstrap = disableBootstrap
+      )
     } else {
-      val biasTraining = oobErrors.map { case (f, e, u) =>
-        // Math.E is only statistically correct.  It should be actualBags / Nib.transpose(i).count(_ == 0)
-        // Or, better yet, filter the bags that don't include the training example
-        val bias = Math.max(Math.abs(e) - u * ratio, 0)
-        (f, bias)
-      }
-      Async.canStop()
-      val biasModel = biasLearner.get.train(biasTraining).getModel().asInstanceOf[Model[PredictionResult[Double]]]
-      Async.canStop()
-
-      if (isRegression) {
-        new BaggedTrainingResult[Double](models.asInstanceOf[ParSeq[Model[PredictionResult[Double]]]], averageImportance, Nib, trainingData, useJackknife, Some(biasModel), ratio, disableBootstrap)
-      } else {
-        new BaggedTrainingResult[Any](models, averageImportance, Nib, trainingData, useJackknife, None, ratio, disableBootstrap)
-      }
+      new BaggedTrainingResult[Any](
+        models = models,
+        featureImportance = averageImportance,
+        Nib = Nib,
+        trainingData = trainingData,
+        useJackknife = useJackknife,
+        biasModel = biasModel,
```

Comment:
Oh I just realized this. `biasModel` will always be None in this branch. Should we just explicit set it `None` ?

Reply:
I thought about that, but decided it would be bug prone because it smears the "what's the bias model" logic out into two places.